core/src/test/scala/integration/kafka/api/ConsumerBounceTest.scala

```diff
@@ -483,7 +483,8 @@ class ConsumerBounceTest extends AbstractConsumerTest with Logging {
       def onPartitionsRevoked(partitions: Collection[TopicPartition]): Unit = {
       }})
     consumer.poll(time.Duration.ofSeconds(3L))
-    assertTrue(assignSemaphore.tryAcquire(1, TimeUnit.SECONDS), "Assignment did not complete on time")
+    TestUtils.waitUntilTrue(() => assignSemaphore.tryAcquire(100, TimeUnit.MILLISECONDS),
+      "Assignment did not complete on time")
```

Comment:
Could the `ConsumerRebalanceListener` be called if we don't poll?

Reply:
Sorry for the late reply, the `onPartitionsAssigned` is only called in `AbstractCoordinator.onJoinComplete`, we should call poll to invoke join `onJoinComplete`.

Reply:
Make sense. So, if `assignSemaphore` is not released after `poll` is executed, it will never be released, right? Should we include `poll` within `waitUntilTrue`?

Reply:
That make sense, I tried to print the time cost by `consumer.poll`, 
```
    println("start:" + System.currentTimeMillis() )
    consumer.poll(time.Duration.ofSeconds(3L))
    println("end:" + System.currentTimeMillis() )
```

and I found that `poll` will cost 3 seconds since no data is consumed before timeout.
```
start:1614844034207
end:1614844037213
```

We can save almost 3 seconds after moving `poll` to `waitUntilTrue` .